Reuse RTCEncodedSourceProducer in RTCRtpScriptTransformer

youennf opened the pull request:
#### fc07c80ec38e0558f2b89431b185d4c4b5468796
<pre>
Reuse RTCEncodedSourceProducer in RTCRtpScriptTransformer
<a href="https://rdar.apple.com/164876812">rdar://164876812</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302635">https://bugs.webkit.org/show_bug.cgi?id=302635</a>

Reviewed by Jean-Yves Avenard.

We do a refactoring to share more code between RTCEncodedSourceProducer and RTCRtpScriptTransformer.
Both were creating readable and writable streams to expose audio/video encoded frames.
With the patch, RTCRtpScriptTransformer is now using RTCEncodedSourceProducer to create/manager readable and writable streams.
We update RTCEncodedSourceProducer to handle key frame generation and logging previously done in RTCRtpScriptTransformer.

Canonical link: <a href="https://commits.webkit.org/303238@main">https://commits.webkit.org/303238@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e98322e723bbb14faccc64cb48e75774ddae246

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131615 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3947 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42580 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139124 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83410 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a95a27b6-62b3-4041-9d09-45c8930ab6bc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3977 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3788 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100480 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68053 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2f531e55-a022-4203-9be7-4e852a1bcc15) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134561 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2846 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117804 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81288 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/a321834a-a3c9-4904-be13-846d5de6779d) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2761 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/511 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82315 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111387 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35930 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141769 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3691 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36446 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108854 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3739 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3271 "Too many flaky failures: fast/images/individual-animation-toggle.html, imported/w3c/web-platform-tests/css/css-contain/content-visibility/content-visibility-auto-text-fragment.html, imported/w3c/web-platform-tests/mediacapture-streams/overconstrained_error.https.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/align_end.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/align_start.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/evil/size_90.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/color_hex.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_normal_wrapped.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_pre.html, imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue_function/class_object/class_namespace.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109097 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27673 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2797 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114133 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56892 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3752 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32534 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3579 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67163 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3834 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3682 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/WebCore/Modules/mediastream/RTCRtpScriptTransformer.h:
```diff
@@ -91,23 +88,13 @@ class RTCRtpScriptTransformer
     RefPtr<RTCRtpTransformBackend> m_backend;
     RefPtr<PendingActivity<RTCRtpScriptTransformer>> m_pendingActivity;
 
     Deque<Ref<DeferredPromise>> m_pendingKeyFramePromises;
```
this isn't use as a Deque; a Vector will do.

youennf replied:
fixed

jyavenard reviewed (approved):
LGTM. Typo in the commit log 
s/exopse/expose

youennf committed:
Reuse RTCEncodedSourceProducer in RTCRtpScriptTransformer
rdar://164876812
https://bugs.webkit.org/show_bug.cgi?id=302635

Reviewed by Jean-Yves Avenard.

We do a refactoring to share more code between RTCEncodedSourceProducer and RTCRtpScriptTransformer.
Both were creating readable and writable streams to expose audio/video encoded frames.
With the patch, RTCRtpScriptTransformer is now using RTCEncodedSourceProducer to create/manager readable and writable streams.
We update RTCEncodedSourceProducer to handle key frame generation and logging previously done in RTCRtpScriptTransformer.

Canonical link: https://commits.webkit.org/303238@main
